### ISSUE TITLE
Manage actions cache in Gitea

### ISSUE THREAD

**Issue by RecordingTheSmile** [go-gitea/gitea#26403]
### Feature Description

We can now utilize caching in all Actions through `actions/cache`. However, Gitea still lacks caching management functionality, meaning it's not possible to view current cache keys or delete caches within Gitea. 

It would be great if Gitea could implement a cache management feature similar to what is available on GitHub.

**Comment by VAllens** [comment]
+10000000000000000000000

**Comment by sebthom** [comment]
I think the issue is not only about cache management via the UI. It seems like there is currently also no way to programmatically delete caches. In GHA I can delete a cache in a build step like this: https://docs.github.com/en/rest/actions/cache?apiVersion=2022-11-28#delete-github-actions-caches-for-a-repository-using-a-cache-key but looking at the Swagger API of Gitea there is no equivalent.

**Comment by RecordingTheSmile** [comment]
> I think the issue is not only about cache management via the UI. It seems like there is currently also no way to programmatically delete caches. In GHA I can delete a cache in a build step like this: https://docs.github.com/en/rest/actions/cache?apiVersion=2022-11-28#delete-github-actions-caches-for-a-repository-using-a-cache-key but looking at the Swagger API of Gitea there is no equivalent.

Agree. And Gitea uses local data storage instead of common C/S database (like MySQL), which makes it impossible to remove cache files manually.

**Comment by lunny** [comment]
> > I think the issue is not only about cache management via the UI. It seems like there is currently also no way to programmatically delete caches. In GHA I can delete a cache in a build step like this: https://docs.github.com/en/rest/actions/cache?apiVersion=2022-11-28#delete-github-actions-caches-for-a-repository-using-a-cache-key but looking at the Swagger API of Gitea there is no equivalent.
> 
> Agree. And Gitea uses local data storage library instead of common C/S database (like MySQL), which makes it impossible to remove cache files manually.

We can have a protocl to ask runners to remove the cache.

**Comment by vjm** [comment]
is there any more information on this? or am i better off using `RUNNER_TOOL_CACHE` instead?

**Comment by lunny** [comment]
`RUNNER_TOOL_CACHE` is different from `actions/cache`. They are two kinds of cache.

**Comment by vjm** [comment]
Yes, correct -- I am not sure that I gain any benefits from `actions/cache` that I am not able to achieve in the local cache as of now, until actions cache allows for more granular management -- is that right? What would be a good example use case where I should use `actions/cache`? I am finding that I can use local cache for docker, npm, python, etc...

**Comment by silverwind** [comment]
Some research on GitHub's caches for things like `actions/cache` and `setup-go` etc:

- 7-day TTL — caches not accessed within 7 days are automatically removed
- 10 GB per-repo limit (can be increased with billing)
- LRU eviction — when the limit is hit, least-recently-accessed caches are evicted first
- Eviction checks run hourly

I think ultimately the cache eviction process belongs in https://gitea.com/gitea/act_runner because it's the runner that creates these caches.

**Comment by silverwind** [comment]
Checking further, `act` (and in turn `act_runner`) already has a hardcoded cache eviction:

https://gitea.com/gitea/act/src/commit/28740d778803ca97663e6126d91e7823f95b3a40/pkg/artifactcache/handler.go#L411-L416

I opened https://gitea.com/gitea/act_runner/issues/807 to make these configurable. Having cache management on the Gitea UI might be good, but I'm personally fine with just configurability.

**Comment by RecordingTheSmile** [comment]
> Some research on GitHub's caches for things like `actions/cache` and `setup-go` etc:
> 
>     * 7-day TTL — caches not accessed within 7 days are automatically removed
> 
>     * 10 GB per-repo limit (can be increased with billing)
> 
>     * LRU eviction — when the limit is hit, least-recently-accessed caches are evicted first
> 
>     * Eviction checks run hourly
> 
> 
> I think ultimately the cache eviction process belongs in https://gitea.com/gitea/act_runner because it's the runner that creates these caches.

GitHub Cache has a web UI for management, and users can manually delete unnecessary caches via the API. This is also a feature that Gitea lacks. I think that after implementing cache management in act_runner, we should also add the same cache management API to Gitea.

**Comment by silverwind** [comment]
Yeah a cache management UI is a different concern than mine and I see this issue is about that, so I am sorry for the noise.
